components/UI/searchBar.tsx

```diff
@@ -1,68 +1,231 @@
+
+const CustomTextField = styled(TextField)(({ theme }) => ({
+  "& .MuiOutlinedInput-root": {
+    padding: "10px 35px",
+    caretColor: "#454545",
+    cursor: "pointer",
+    "& fieldset": {
+      border: "1px solid #CDCCCC",
+      borderRadius: "20px",
+      boxShadow: "0px 2px 30px 0px rgba(0, 0, 0, 0.06)",
+      backgroundColor: "#FFFFFF",
+      cursor: "pointer",
+    },
+    "& .MuiInputBase-input": {
+      color: "#454545",
+      fontSize: "24px",
+      fontStyle: "normal",
+      fontWeight: "700",
+      lineHeight: "24px",
+      letterSpacing: "0.24px",
+      textAlign: "center",
+      zIndex: "1",
+      cursor: "pointer",
+    },
+    "&:hover fieldset": {
+      border: "1px solid #CDCCCC",
+    },
+    "& ::placeholder": {
+      color: "#B0AEAE",
+      textAlign: "center",
+      fontSize: "24px",
+      fontStyle: "normal",
+      fontWeight: "700",
+      lineHeight: "24px",
+      letterSpacing: "0.24px",
+      justifyContent: "center",
+      alignItems: "center",
+    },
+    "&.Mui-focused ::placeholder": {
+      color: "transparent",
+    },
+    "&.Mui-focused fieldset": {
+      borderColor: theme.palette.primary.main,
+    },
+  },
+}));
 
 type SearchBarProps = {
   onChangeTypedValue?: (typedValue: string) => void;
+  showHistory: boolean;
 };
+
 const SearchBar: FunctionComponent<SearchBarProps> = ({
   onChangeTypedValue,
+  showHistory,
 }) => {
   const router = useRouter();
+  const resultsRef = useRef<HTMLDivElement>(null);
   const [typedValue, setTypedValue] = useState<string>("");
   const isDomainValid = useIsValid(typedValue);
+  const [currentResult, setCurrentResult] = useState<SearchResult | null>();
+  const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+  const [showResults, setShowResults] = useState<boolean>(false);
+  const { provider } = useContext(StarknetIdJsContext);
+  const contract = new Contract(
+    naming_abi as Abi,
+    process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
+    provider as Provider
+  );
+
+  useEffect(() => {
+    const existingResults =
+      JSON.parse(localStorage.getItem("search-history") as string) || [];
+    const firstResults = existingResults.slice(0, 5);
+    const resultPromises = firstResults.map(
+      (result: { name: string; lastAccessed: number }) =>
+        getStatus(result.name, result.lastAccessed)
+    );
+    Promise.all(resultPromises).then((fullResults) => {
+      fullResults.sort(
+        (firstResult: SearchResult, secondResult: SearchResult) =>
+          secondResult.lastAccessed - firstResult.lastAccessed
+      );
+      setSearchResults(fullResults);
+    });
+  }, []);
+
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        resultsRef.current &&
+        !resultsRef.current.contains(event.target as Node)
+      ) {
+        setShowResults(false);
+      } else {
+        setShowResults(true);
```

Comment:
if you click inside, shouldn't it be already toggled?

Reply:
I'm not sure I understand the question, you mean I should only update the state if it has not already the right value ?